src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs

```diff
@@ -68,23 +69,26 @@ public void CalculatePKAnalyses(IPopulationDataCollector populationDataCollector
          //Calculate first global PK that should always be updated
          _globalPKAnalysisPresenter.CalculatePKAnalysis(new[] { simulation });
 
-         var pkParameters = extractPKParameters(populationAnalysis, simulation);
-         if (!pkParameters.Any())
+         var pkParametersCache = extractPKParameters(populationAnalysis, simulation);
+         if (!pkParametersCache.Any())
             return;
 
-         var captionPrefix = populationAnalysis.AllFieldNamesOn(PivotArea.DataArea);
-
-         var pkAnalyses = _pkAnalysesTask.AggregatePKAnalysis(simulation, pkParameters, populationAnalysis.SelectedStatistics, captionPrefix[0]);
-         _allPKAnalysesIndividualPKValues.AddRange(pkAnalyses);
+         pkParametersCache.KeyValues.Each(pkParameters =>
+         {
+            var pkAnalyses = _pkAnalysesTask.AggregatePKAnalysis(simulation, pkParameters.Value, populationAnalysis.SelectedStatistics, pkParameters.Key);
+            _allPKAnalysesIndividualPKValues.AddRange(pkAnalyses);
+         });
 
          updateView();
       }
 
-
-      private IReadOnlyList<QuantityPKParameter> extractPKParameters(PopulationStatisticalAnalysis populationAnalysis, PopulationSimulation populationSimulation)
+      private Cache<string, IReadOnlyList<QuantityPKParameter>> extractPKParameters(PopulationStatisticalAnalysis populationAnalysis, PopulationSimulation populationSimulation)
       {
-         var fields = populationAnalysis.AllFields.OfType<PopulationAnalysisOutputField>().Select(x => x.QuantityPath);
-         return fields.SelectMany(x => populationSimulation.PKAnalyses.AllPKParametersFor(x)).ToList();
+         var cache = new Cache<string, IReadOnlyList<QuantityPKParameter>>();
+         var fields = populationAnalysis.AllFields.OfType<PopulationAnalysisOutputField>().ToList();
```

Comment:
What about you call it output? Field is the generic term that means anything from parameters to pk etc..

Reply:
You mean outputFields or something like that. This isn't the output of the method.

Reply:
output is part of the "domain" of PKSim. output of the simulation. outputField is good too. I did not know what field was in the context at first glance